intellij/src/saros/intellij/eventhandler/filesystem/LocalFilesystemModificationHandler.java

```diff
@@ -259,11 +287,13 @@ private void generateResourceCreationActivity(@NotNull VirtualFileEvent virtualF
       activity =
           new FileActivity(
               user, Type.CREATED, FileActivity.Purpose.ACTIVITY, path, null, content, charset);
-
-      editorManager.openEditor(path, false);
     }
 
     dispatchActivity(activity);
+
+    if (!createdVirtualFile.isDirectory()) {
+      editorManager.openEditor(path, false);
```

Comment:
I still does not understand why the editor is opened here. Is the open call a "connect" call so we now that we now must track changes or it is really opened in the UI ?

Reply:
It is both. To only open an editor, [`ProjectAPI.openEditor(...)`](https://github.com/saros-project/saros/blob/master/intellij/src/saros/intellij/editor/ProjectAPI.java#L59) can be used.
The methods `LocalEditorManager.openEditor(...)` and `LocalEditorHandler.openEditor(...)` both open the editor and set it up for Saros (adding it to the editor pool and registering handlers; the handler registration gets removed with 1156cb640cbb7b6d3227abb3101882945f07e4a1 as part of #446).

The problem is that I am not aware of a method that gives us arbitrary editor objects for open editors. The method described [here](https://www.jetbrains.org/intellij/sdk/docs/tutorials/editor_basics/working_with_text.html#getting-an-instance-of-the-active-editor) can give us the active editor, but that is about it.

So, to add the editor to the editor pool, we either have to use `ProjectAPI.openEditor(...)`, which automatically opens an editor for the given VirtualFile if it is not already open, or wait until the editor is opened so that we can get the object from the listener.

Reply:
An approach to remove this duality would be to remove the editor pool and always resolve the resource for the editor directly if necessary. Will look into whether this is plausible.

Reply:
Removing the EditorPool seems plausible, but it is somewhat of a hassle as it is used a lot in `EditorManager`. I don't think it is currently worth it to invest time into removing it.